### PR TITLE
Populate the source artifact field from PyPI metadata

### PR DESCRIPTION
**Pull request by haikoschol**
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/256)

### CODE REVIEW

**Comment by sschuberth** [comment]
> PIP: Prefer wheels for binary artifacts

Why? Could you please add a rationale to the commit message?

File: analyzer/src/main/kotlin/managers/PIP.kt [hunk 1]
```diff
@@ -307,6 +307,14 @@ class PIP : PackageManager() {
         )
     }
 
+    private fun getSourceArtifact(pkgReleases: ArrayNode): RemoteArtifact {
+        val pkgSource = pkgReleases.asSequence().find { it["packagetype"].asText() == "sdist" } ?: return RemoteArtifact.EMPTY
+        val url = pkgSource["url"]?.asText() ?: return RemoteArtifact.EMPTY
+        val hash = pkgSource["md5_digest"]?.asText() ?: return RemoteArtifact.EMPTY
+
+        return RemoteArtifact(url = url, hash = hash, hashAlgorithm = HashAlgorithm.MD5)
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/PIP.kt]: Maybe in this case, as it's quite obvious, consider dropping the named arguments for shortness.

**Comment by sschuberth** [comment]
FYI, CI fails due to

    PIP.kt:310:71: Exceeded max line length (120)

**Comment by haikoschol** [comment]
It just occurred to me that I forgot to prefer `tar.bz2` for the source artifact.

**Comment by sschuberth** [comment]
> It just occurred to me that I forgot to prefer `tar.bz2` for the source artifact.

I believe this can be nicely done in a follow-up commit, either in this or a new PR, as you prefer.

**Comment by mnonnenmacher** [comment]
@haikoschol : Before you prioritize `tar.bz2` we need to add support for this format in `ArchiveUtils.kt`. Could you do this in a separate PR, too?

**Pull request merged**